packages/dev/core/src/Meshes/Node/Blocks/linearClonerBlock.ts

```diff
@@ -0,0 +1,213 @@
+import { NodeGeometryBlock } from "../nodeGeometryBlock";
+import type { NodeGeometryConnectionPoint } from "../nodeGeometryBlockConnectionPoint";
+import { RegisterClass } from "../../../Misc/typeStore";
+import { NodeGeometryBlockConnectionPointTypes } from "../Enums/nodeGeometryConnectionPointTypes";
+import { Matrix, Quaternion, Vector3 } from "../../../Maths/math.vector";
+import { type NodeGeometryBuildState } from "../nodeGeometryBuildState";
+import type { VertexData } from "core/Meshes/mesh.vertexData";
+import { type Nullable } from "core/types";
+import { PropertyTypeForEdition, editableInPropertyPage } from "core/Decorators/nodeDecorator";
+
+/**
+ * Block used to clone geometry along a line
+ */
+export class LinearClonerBlock extends NodeGeometryBlock {
+    private _vertexData: VertexData;
+    private _currentIndex: number;
+    private _currentLoopIndex: number;
+
+    /**
+     * Gets or sets a boolean indicating that this block can evaluate context
+     * Build performance is improved when this value is set to false as the system will cache values instead of reevaluating everything per context change
+     */
+    @editableInPropertyPage("Evaluate context", PropertyTypeForEdition.Boolean, "ADVANCED", { notifiers: { rebuild: true } })
+    public evaluateContext = true;
+
+    /**
+     * Gets the current index in the current flow
+     * @returns the current index
+     */
+    public getExecutionIndex(): number {
+        return this._currentIndex;
+    }
+
+    /**
+     * Gets the current face index in the current flow
+     * @returns the current face index
+     */
+    public getExecutionFaceIndex(): number {
+        return 0;
+    }
+
+    /**
+     * Gets the current loop index in the current flow
+     * @returns the current loop index
+     */
+    public getExecutionLoopIndex(): number {
+        return this._currentLoopIndex;

```

Comment:
in your case this is this._currentIndex;

Reply:
It was telling me I needed this?